lib/net.c

```diff
@@ -838,10 +838,43 @@ check_port_mtu(struct gatekeeper_if *iface, unsigned int port_idx,
 	return 0;
 }
 
+static int
+check_port_cksum(struct gatekeeper_if *iface, unsigned int port_idx,
+	const struct rte_eth_dev_info *dev_info,
+	struct rte_eth_conf *port_conf, bool gatekeeper_server)
+{
+	if (ipv4_if_configured(iface) &&
+			(port_conf->txmode.offloads &
+			 DEV_TX_OFFLOAD_IPV4_CKSUM) &&
+			!(dev_info->tx_offload_capa &
+			DEV_TX_OFFLOAD_IPV4_CKSUM)) {
+		G_LOG(NOTICE, "net: port %hu (%s) on the %s interface doesn't support offloading IPv4 checksumming\n",
+			iface->ports[port_idx], iface->pci_addrs[port_idx],
+			iface->name);
+		port_conf->txmode.offloads &= ~DEV_TX_OFFLOAD_IPV4_CKSUM;
+		return -1;
+	}
+
+	if (!gatekeeper_server &&
+			(port_conf->txmode.offloads &
+			DEV_TX_OFFLOAD_UDP_CKSUM) &&
+			!(dev_info->tx_offload_capa &
+			DEV_TX_OFFLOAD_UDP_CKSUM)) {
+		G_LOG(NOTICE, "net: port %hu (%s) on the %s interface doesn't support offloading UDP checksumming\n",
+			iface->ports[port_idx], iface->pci_addrs[port_idx],
+			iface->name);
+		port_conf->txmode.offloads &= ~DEV_TX_OFFLOAD_UDP_CKSUM;
+		return -1;
+	}
+
+	return 0;
+}
+
 static int
 check_port_offloads(struct gatekeeper_if *iface,
 	struct rte_eth_conf *port_conf)
 {
+	struct net_config *net_conf = get_net_conf();
```

Comment:
We can use variable ``config`` for network configration directly within ``lib/net.c``